processor/src/decorators/mod.rs

```diff
@@ -8,10 +8,16 @@ impl Process {
     pub(super) fn execute_decorator(
         &mut self,
         decorator: &Decorator,
+        in_debug_mode: bool,
```

Comment:
instead of passing `in_debug_mode()` here, could we not just do `self.decoder.in_debug_mode()` inside the function?

Maybe it would actually make sense to add `in_debug_mode` field to the `Process` struct - but we can probably leave this for future PRs.

Reply:
You're right, thanks, fixed.